govcd/disk.go

```diff
@@ -339,3 +348,73 @@ func (vdc *Vdc) QueryDisk(diskName string) (DiskRecord, error) {
 
 	return *newDisk, nil
 }
+
+// GetDiskByHref finds a Disk by HREF
+// On success, returns a pointer to the Disk structure and a nil error
+// On failure, returns a nil pointer and an error
+func (vdc *Vdc) GetDiskByHref(diskHref string) (*Disk, error) {
+	util.Logger.Printf("[TRACE] Get Disk By Href: %s\n", diskHref)
+	Disk := NewDisk(vdc.client)
+
+	_, err := vdc.client.ExecuteRequest(diskHref, http.MethodGet,
+		"", "error retrieving Disk: %s", nil, Disk.Disk)
+	if err != nil {
+		return nil, err
+	}
+	return Disk, nil
+}
+
+// GetDisksByName finds one or more Disks by Name
+// On success, returns a pointer to the Disk list and a nil error
+// On failure, returns a nil pointer and an error
+func (vdc *Vdc) GetDisksByName(diskName string, refresh bool) (*[]Disk, error) {
+	util.Logger.Printf("[TRACE] Get Disk By Name: %s\n", diskName)
+	var diskList []Disk
+	if refresh {
+		err := vdc.Refresh()
+		if err != nil {
+			return nil, err
+		}
+	}
+	for _, resourceEntities := range vdc.Vdc.ResourceEntities {
+		for _, resourceEntity := range resourceEntities.ResourceEntity {
+			if resourceEntity.Name == diskName && resourceEntity.Type == "application/vnd.vmware.vcloud.disk+xml" {
+				disk, err := vdc.GetDiskByHref(resourceEntity.HREF)
+				if err != nil {
+					return nil, err
+				}
+				diskList = append(diskList, *disk)
+			}
+		}
+	}
+	if len(diskList) == 0 {
+		return nil, ErrorEntityNotFound
+	}
+	return &diskList, nil
+}
+
+// GetDiskById finds a Disk by ID
+// On success, returns a pointer to the Disk structure and a nil error
+// On failure, returns a nil pointer and an error
+func (vdc *Vdc) GetDiskById(diskId string, refresh bool) (*Disk, error) {
+	util.Logger.Printf("[TRACE] Get Disk By Id: %s\n", diskId)
+	if refresh {
+		err := vdc.Refresh()
+		if err != nil {
+			return nil, err
+		}
+	}
+	diskHREF := vdc.client.VCDHREF
```

Comment:
Either I got it wrong, either this var name implies some readability problem. Is this really a diskHREF stored? Is it not becoming a diskHREF below in line 418 (`diskHREF.Path += fmt.Sprintf("/disk/%s", DiskBareId)`)

Reply:
Moved to better place